docs: add Electron 27 blog post (#454)
electron/website: blog/electron-27-0.md

---
title: Electron 27.0.0
date: 2023-10-10T00:00:00.000Z
authors:
  - name: vertedinde
    url: 'https://github.com/vertedinde'
    image_url: 'https://github.com/vertedinde.png?size=96'
slug: electron-27-0
---

> Electron 27.0.0 has been released! It includes upgrades to Chromium `118.0.5993.32`, V8 `11.8`, and Node.js `18.17.1`. Read below for more details!

Reviewer: ```suggestion
Electron 27.0.0 has been released! It includes upgrades to Chromium `118.0.5993.32`, V8 `11.8`, and Node.js `18.17.1`.
```

Personally I'd drop the "Read below for more details!", since it's implied in a blog post and doesn't (at least to me) add obviously useful information.

---

The Electron team is excited to announce the release of Electron 27.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://releases.electronjs.org/releases/stable). Continue reading for details about this release.

> If you have any feedback, please share it with us on Twitter, or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

Reviewer: ```suggestion
If you have any feedback, please share it with us on [Twitter](https://twitter.com/electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).
```

Author: Actually wondering if we should maybe add Mastodon/BlueSky here too? 🤔

## Notable Changes

### Stack Changes

- Chromium `118.0.5993.32`
  - [New in Chrome 118](https://developer.chrome.com/blog/new-in-chrome-118/)
  - [New in DevTools 118](https://developer.chrome.com/blog/new-in-devtools-118/)
- Node.js `18.17.1`

>   - [Node 18.17.1 blog post](https://nodejs.org/en/blog/release/v18.17.1/)

Reviewer: Potentially worth including any other Node.js patches or minors since 18.6.1, which is what was included in 26.0.0. IMO at the very least, 18.17.0 would probably be a more interesting link than 18.17.1.

Author: Good point, added the 18.17.0 post - looks like there weren't any patches between 18.16.1 and 18.17.0, so that was easy 😄 We can include patches & minor in the future too, it's a good idea

- V8 `11.8`

### Breaking Changes

### Removed: Extended support for Windows 7/8/8.1

> Electron 22 is now end of life (see [Farewell, Windows 7/8/8.1](https://www.electronjs.org/blog/windows-7-to-8-1-deprecation-notice) for more details). All currently supported versions of Electron have dropped support for Windows 7/8/8.1.

Reviewer 2: I'm not sure if this should be in a different section, this skim reads as "Electron 27 dropped support", moving it into the "end of extended support" section at the bottom might be appropriate

Author: Yeah, I was waffling on whether or not we should mention it here - removed here and left in the end of extended support section at the end 🙂

### Removed: macOS 10.13 / 10.14 support

macOS 10.13 (High Sierra) and macOS 10.14 (Mojave) are no longer supported by [Chromium](https://chromium-review.googlesource.com/c/chromium/src/+/4629466).

Older versions of Electron will continue to run on these operating systems, but macOS 10.15 (Catalina)
or later will be required to run Electron v27.0.0 and higher.

### Deprecated: `ipcRenderer.sendTo()`

> The `ipcRenderer.sendTo()` API has been deprecated. It should be replaced by setting up a [`MessageChannel`](tutorial/message-ports.md#setting-up-a-messagechannel-between-two-renderers) between the renderers.

Reviewer 3: ```suggestion
The `ipcRenderer.sendTo()` API has been deprecated. It should be replaced by setting up a [`MessageChannel`](https://www.electronjs.org/docs/latest/tutorial/message-ports#setting-up-a-messagechannel-between-two-renderers) between the renderers.
```

// Replace with
nativeTheme.shouldUseDarkColors;

// Removed
systemPreferences.setAppLevelAppearance('dark');
// Replace with
nativeTheme.themeSource = 'dark';
```

### Removed: `alternate-selected-control-text` value for `systemPreferences.getColor`

The `alternate-selected-control-text` value for `systemPreferences.getColor`
has been removed. Use `selected-content-background` instead.

```js
// Removed
systemPreferences.getColor('alternate-selected-control-text');
// Replace with
systemPreferences.getColor('selected-content-background');
```

### New Features

- Added app accessibility transparency settings api [#39631](https://github.com/electron/electron/pull/39631)
- Added support for `chrome.scripting` extension APIs [#39675](https://github.com/electron/electron/pull/39675)
- Enabled `WaylandWindowDecorations` by default [#39644](https://github.com/electron/electron/pull/39644)

## End of Support for 24.x.y

Electron 24.x.y has reached end-of-support as per the project's [support policy](https://www.electronjs.org/docs/latest/tutorial/electron-timelines#version-support-policy). Developers and applications are encouraged to upgrade to a newer version of Electron.

> | E27 (Aug'23) | E28 (Oct'23) | E29 (Jan'24) |

Reviewer 3: I think the dates in this header need to be updated.

Author: D'oh they sure do, thanks for the catch, will update those
